---
title: "ERGO-ii analysis"
execute:
  echo: false
---

## How to run **ERGO-ii anlaysis** in the computational infrastructure of INAB

### Prerequisites

#### Install R

Follow instructions here: [The Comprehensive R Archive Network](https://cloud.r-project.org/). If you cannot download R on your local machine please contact [inab-support\@certh.gr](mailto:inab-support@certh.gr).

### Install RStudio

Follow instructions here: [RStudio Desktop - Posit](https://posit.co/download/rstudio-desktop/) (*2: Install RStudio*). If you cannot download R on your local machine please contact [inab-support\@certh.gr](mailto:inab-support@certh.gr).\

### Organize your data with the following structure

Your data should be organized with the following format:

my_data/

├─ Group1/

│  ├─ Clonotypes/

│  │  ├─ Clonotypes_sample1.txt

│  │  ├─ Clonotypes_sample2.txt

│  │  ├─ ...

│  ├─ peptides.xlsx

├─ Group2/

│  ├─ Clonotypes/

│  │  ├─ ...

│  ├─ peptides.xlsx

├─ .../

│  ├─ .../

│  ├─ ...

### Peptides file

The peptides file is a 3 – column table containing the following information:

| HLA allele   | Peptide    | Sample  |
|--------------|------------|---------|
| HLA-A\*02:02 | AMAIAVPANV | Sample1 |
| HLA-A\*02:02 | MAIAVPANV  | Sample1 |
| HLA-A\*03:01 | AIAVPANVY  | Sample2 |
| HLA-C\*07:01 | QSYSPPFSF  | Sample3 |

### Steps to run RNAseq analysis:

1.  Log in to the appropriate server:

    o 160.40.71.4

2.  Identify where your NGS results (.fastq files) are located:

    o e.g., /mnt/new_home/stamatopoulos_research/2025_05_21_ERIC_study

3.  Move to the your working directory:

    ``` bash
    cd /work/folder_of_each_user
    ```

4.  Create a new analysis directory:

    ``` bash
    mkdir RNAseq_analysis_2025_05_21
    ```

5.  Move to the your new analysis directory:

    ``` bash
    cd /work/RNAseq_analysis_2025_05_21
    ```

6.  Download script to your analysis directory

    ``` bash
    wget https://raw.githubusercontent.com/natanast/RNAseq_analysis/main/bash/star.sh
    ```

    ``` bash
    wget https://raw.githubusercontent.com/natanast/RNAseq_analysis/main/bash/runSTAR.sh 
    ```

7.  Create a SampleList (you have to update with the right path to your fastq files), for example:

    ``` bash
    ls -la /mnt/new_home/stamatopoulos_research/RNAseq_2025_05_21 | awk '{print $9}' | awk -F "_R" '{print $1}' | sort | uniq > SampleList
    ```

8.  Remove samples that should not be analyzed:

    ``` bash
    nano SampleList
    ```

    In order to **edit** the SampleList use the following:

    o `Delete button`: Removes one letter

    o `Ctrl + k`: Removes a whole row

    o `Ctrl + o and Enter`: Saves the modified SampleList

    o `Ctrl + x`: Exits and returns to the terminal

9.  Run the analysis (you have to **update** with the right path):

    ``` bash
    nohup bash runSTAR.sh \
    /mnt/new_home/stamatopoulos_research/RNAseq_2025_05_21 \
    SampleList \
    /mnt/new_home/bio_tmp/HSapRefData/hg38/gencode/star \
    /mnt/new_home/bio_tmp/HSapRefData/hg38/gencode/gencode.v47.primary_assembly.basic.annotation.gtf \
    16 \
    &
    ```

10. After the analysis is completed, move **your folder** from /work to /mnt/new_home

    ``` bash
    cd ..
    ```

    ``` bash
    rsync -r \
    RNAseq_analysis_2025_05_21 \
    /mnt/new_home/stamatopoulos_research
    ```
